include/mrdox/Metadata/Template.hpp

```diff
@@ -14,39 +14,254 @@
 #define MRDOX_METADATA_TEMPLATE_HPP
 
 #include <mrdox/Platform.hpp>
-#include <mrdox/Metadata/TemplateArg.hpp>
-#include <mrdox/Metadata/TemplateParam.hpp>
+#include <mrdox/ADT/Optional.hpp>
+#include <mrdox/Metadata/Type.hpp>
+#include <optional>
+#include <string>
+#include <vector>
 
 namespace clang {
 namespace mrdox {
 
+enum class TParamKind : int
+{
+    None = 0, // for bitstream
+    // template type parameter, e.g. "typename T" or "class T"
+    Type,
+    // template non-type parameter, e.g. "int N" or "auto N"
+    NonType,
+    // template template parameter, e.g. "template<typename> typename T"
+    Template,
+};
+
+struct TParam;
+
+struct TypeTParam
+{
+    /** Default type for the type template parameter */
+    std::optional<TypeInfo> Default;
```

Comment:
unique_ptr?

Reply:
This should remain as an optional since `TypeInfo` is just a `Reference`. We can implement `OptionalTypeInfo` using `Optional` later on